test/multisigwallet.test.ts

```diff
@@ -0,0 +1,581 @@
+import { assert } from 'chai';
+
+import {
+  ContractContextDefinition,
+  Method,
+  TransactionLog,
+  TransactionResult
+} from 'truffle';
+import * as Web3 from 'web3';
+
+import {
+  ConfirmationEvent,
+  ExecutionEvent,
+  ExecutionFailureEvent,
+  MultiSigWallet,
+  OwnerAdditionEvent,
+  OwnerRemovalEvent,
+  RequirementChangeEvent,
+  RevocationEvent,
+  SubmissionEvent,
+  WalletArtifacts
+} from 'wallet';
+
+import {
+  assertReverts,
+  deployWalletFunction,
+  findLastLog,
+  findLastTransactionId,
+  findNewRequirement,
+  getData
+} from './helpers';
+
+declare const web3: Web3;
+declare const artifacts: WalletArtifacts;
+declare const contract: ContractContextDefinition;
+
+const MultiSigWalletContract = artifacts.require('MultiSigWallet');
+
+contract('MultiSigWallet', accounts => {
+  const owners = [accounts[0], accounts[1], accounts[2]];
+  const required = 3;
+
+  let wallet: MultiSigWallet;
+
+  beforeEach(async () => {
+    wallet = await MultiSigWalletContract.new(owners, required, {
+      from: owners[0]
+    });
+    assert.isOk(wallet);
+  });
+
+  describe('#ctor', () => {
+    it('should revert for empty list of owners', async () => {
+      await assertReverts(async () => {
+        await MultiSigWalletContract.new([], 0);
+      });
+    });
+
+    it('should revert for zero required', async () => {
+      await assertReverts(async () => {
+        await MultiSigWalletContract.new(owners, 0);
+      });
+    });
+
+    it('should create wallet', async () => {
+      assert.ok(wallet);
+    });
+
+    it('should set required', async () => {
+      assert.equal(await wallet.required(), required);
+    });
+
+    it('should set owners', async () => {
+      assert.deepEqual(await wallet.getOwners(), owners);
+    });
+
+    it('should not have any transactions', async () => {
+      assert.equal(await wallet.getTransactionCount(true, true), 0);
+    });
+  });
+
+  describe('#submitTransaction', () => {
+    let tx: TransactionResult;
+
+    context('with invalid data', async () => {
+      it('should revert for non-existing owner', async () => {
```

Comment:
Replace `non-existing owner` with `non-owner`.

Reply:
Futhermore, the test case is not isolated. It has two reasons to fail. First - called by non-owner, Second - invalid data.